backend-test/Controllers/ProgramControllerTests.hs

```diff
@@ -0,0 +1,54 @@
+module Controllers.ProgramControllerTests (
+    test_programController
+) where
+
+import Config (runDb)
+import Control.Monad.IO.Class (liftIO)
+import Controllers.Program (index)
+import qualified Data.ByteString.Lazy.Char8 as BL
+import qualified Data.Text as T
+import Data.Time (getCurrentTime)
+import Database.DataType (PostType (..))
+import Database.Persist.Sqlite (SqlPersistM, insert_)
+import Database.Tables (Post (..))
+import Happstack.Server (rsBody)
+import Test.Tasty (TestTree)
+import Test.Tasty.HUnit (assertEqual, testCase)
+import TestHelpers (clearDatabase, runServerPart, withDatabase)
+
+-- | List of test cases as (label, input courses, expected output)
```

Comment:
All of the comments in this file need to be updated (they refer to courses, not programs)